src/throttler-storage-redis.service.ts

```diff
@@ -25,21 +22,49 @@ export class ThrottlerStorageRedisService implements ThrottlerStorageRedis, OnMo
     }
   }
 
+  /**
+   * @description: This method look for members (records) in a Redis set, see: https://redis.io/docs/data-types/sets/
+   *
+   * @param key
+   */
   async getRecord(key: string): Promise<number[]> {
-    const ttls = (
-      await this.redis.scan(
-        0,
-        'MATCH',
-        `${this.redis?.options?.keyPrefix}${key}:*`,
-        'COUNT',
-        this.scanCount,
-      )
-    ).pop();
-    return (ttls as string[]).map((k) => parseInt(k.split(':').pop())).sort();
+    const setMembers = await this.redis.smembers(key);
+    const now = +new Date();
+
+    /*
+     That is a Redis SET cleaning mechanism.
+     - There is no way to set ttl for each set member separately.
+     - Each time new member is added to the set, set expiration is moving forward.
+     - We need to clean expired members manually (to avoid extra memory usage).
+     - That is needed in cases when set not expires during long period of time (because of continuous addRecord triggers)
+    */
+    const expiredMembers = setMembers.filter((m) => +m < now);
+    if (expiredMembers.length) {
+      const multi = this.redis.multi();
+      for (const expiredMember of expiredMembers) {
+        multi.srem(expiredMember);
+      }
+      await multi.exec();
+    }
+
+    return setMembers
+      .filter((m) => +m > now)
+      .map((m) => +m)
+      .sort();
```

Comment:
Please keep your code clean, don't use ugly javascript hacks such with the `+` operator to parse to an integer. Follow the airbnb styleguide and make sure to use type hints inside of the functions.

```suggestion
    const expiredMembers = setMembers.filter((m: string) => parseInt(m) < now);
    if (expiredMembers.length) {
      const multi = this.redis.multi();
      for (const expiredMember of expiredMembers) {
        multi.srem(expiredMember);
      }
      await multi.exec();
    }

    return setMembers
      .filter((m: string) => parseInt(m) > now)
      .map((m: string) => parseInt(m))
      .sort();
```